src/onnx/onnx.cpp

```diff
@@ -1807,6 +1808,39 @@ struct onnx_parser
         return ret_ins;
     }
 
+    instruction_ref
+    parse_onehot(const std::string&, node_info info, std::vector<instruction_ref> args)
+    {
+        migraphx::argument depth_arg = args[1]->eval();
+        check_arg_empty(depth_arg, "ONEHOT: depth - dynamic shape not supported");
+        size_t depth = static_cast<size_t>(depth_arg.at<int32_t>());
```

Comment:
Why the `static_cast`? Why not read it as `size_t` from the argument?

Reply:
fixed